packages/editor/src/browser/diff-command.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { SelectionService } from '@theia/core';
+import { OpenerService } from '@theia/core/lib/browser';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { FileSystem } from '@theia/filesystem/lib/common/filesystem';
+import { UriAwareCommandHandler, UriCommandHandler, WorkspaceCommands } from '@theia/workspace/lib/browser';
+import { CommandContribution, CommandRegistry } from '@theia/core/lib/common/command';
+import { DiffUris } from './diff-uris';
+
+@injectable()
+export class DiffCommandContribution implements CommandContribution {
+
+    @inject(FileSystem) protected readonly fileSystem: FileSystem;
+    @inject(MessageService) protected readonly messageService: MessageService;
+    @inject(SelectionService) protected readonly selectionService: SelectionService;
+    @inject(OpenerService) protected readonly openerService: OpenerService;
+
+    registerCommands(registry: CommandRegistry): void {
+        registry.registerCommand(WorkspaceCommands.FILE_COMPARE, this.newMultiUriAwareCommandHandler({
```

Comment:
not sure how we can handle it, maybe move `DIffUris` to core and register the command in the workspace extension, it seems to belong there